src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+        res.push_back(std::make_pair("Boinc Public Key", bb.BoincPublicKey));
+    }
 
-        res["xbbOrganization"]=bb.Organization;
-        res["xbbClientVersion"]=bb.clientversion;
+    return res;
+}
 
-        res["xbbNeuralHash"]=bb.NeuralHash;
-        res["xbbCurrentNeuralHash"]=bb.CurrentNeuralHash;
-        res["xbbNeuralContractSize"]=bb.superblock.length();
-    }
-    else
+std::vector<std::pair<std::string, std::string>> GetTxNormalBoincHashInfo(const CMerkleTx& mtx)
+{
+    assert(!mtx.IsCoinStake() && !mtx.IsCoinBase());
+    std::vector<std::pair<std::string, std::string>> res;
+
+    try
     {
-        if(bb.superblock.length()>=20)
+        const std::string &msg = mtx.hashBoinc;
+
+        res.push_back(std::make_pair("Network Time", TimestampToHRDate((double)mtx.nTime)));
+
+        if (fDebug)
+            res.push_back(std::make_pair("Message Length", ToString(msg.length())));
+
+        std::string sMessageType = ExtractXML(msg, "<MT>", "</MT>");
+        std::string sTxMessage = ExtractXML(msg, "<MESSAGE>", "</MESSAGE>");
+        std::string sRainMessage = ExtractXML(msg, "<NARR>", "</NARR>");
+
+        if (sMessageType.length())
+        {
+            res.push_back(std::make_pair("Message Type", "Contract"));
+
+            if (sMessageType == "beacon")
+            {
+                res.push_back(std::make_pair("Contract Type", "Beacon"));
+
+                std::string sBeaconCommit = ExtractXML(msg, "<MA>", "</MA>");
+
+                if (sBeaconCommit == "A")
+                {
+                    std::string sBeaconCPID = ExtractXML(msg, "<MK>", "</MK>");
+                    std::string sBeaconEncodedContract = ExtractXML(msg, "<MV>", "</MV>");
+
+                    if (sBeaconEncodedContract.length() < 256)
+                    {
+                        // If for whatever reason the contract is not a proper one and the average length does exceed this size; Without this a seg fault will occur on the DecodeBase64
+                        // Another example is if an admin accidently uses add instead of delete in addkey to remove a beacon the 1 in <MV>1</MV> would cause a seg fault as well
+                        res.push_back(std::make_pair("ERROR", "Contract length for beacon is less then 256 in length. Size: " + ToString(sBeaconEncodedContract.length())));
+
+                        if (fDebug)
+                            res.push_back(std::make_pair("Contract Data", sBeaconEncodedContract));
+
+                        return res;
+                    }
+
+                    std::string sBeaconDecodedContract = DecodeBase64(sBeaconEncodedContract);
+                    std::vector<std::string> vBeaconContract = split(sBeaconDecodedContract.c_str(), ";");
+                    std::string sBeaconAddress = vBeaconContract[2];
+                    std::string sBeaconPublicKey = vBeaconContract[3];
+
+                    res.push_back(std::make_pair("Beacon Commit", "Add"));
```

Comment:
`Commit` is actually called Action in rest of code.
Also vBeaconContract[0] contains CPIDv2, which reminds me to remove it from new beacons.

Reply:
yes thus why i didnt add it. just grabbed the cpid MK instead which is just cpid